files_orpg_sw/src/cpc101/lib003/orpgvst.3

orpgvst(3)		ORPG SDD Manpage		orpgvst(3)

NAME

orpgvst - ORPGVST Library Functions.

SYNOPSIS

#include <orpg.h>
#include <orpgvst.h>

char*  ORPGVST_read ( char* vol_stat )
int ORPGVST_io_status ()
int ORPGVST_status_update_flag ()
time_t ORPGVST_status_update_time ()
int ORPGVST_get_volume_flag ()
int ORPGVST_get_volume_duration ()
int ORPGVST_get_volume_date ()
int ORPGVST_get_previous_status ()
int ORPGVST_get_mode ()
int ORPGVST_get_vcp ()
int ORPGVST_get_vcp ()
int ORPGVST_get_volume_scan ()
int ORPGVST_get_scan_count ()
int ORPGVST_get_number_elevations ()
int ORPGVST_get_vcp_id ()
int ORPGVST_get_elevation (int indx)
int ORPGVST_get_index (int indx)
int ORPGVST_get_rda_index (int indx)
unsigned long ORPGVST_get_volume_number ()
unsigned long ORPGVST_get_volume_time ()
int ORPGVST_get_current_vcp ()
int ORPGVST_get_superres_bitmap()
int ORPGVST_get_dual_pol_expected()
int ORPGVST_is_rpg_elev_superres( int rpg_elev_num )
int ORPGVST_get_vcp_supplemental_data()
int ORPGVST_get_n_sails_cuts ()
int ORPGVST_get_sails_cut_seq (int indx)

DESCRIPTION

The ORPGVST family of functions provide an API to reading the Volume Status
message (VOL_STAT_GSM_ID) component of the General Status Message
(ORPGDAT_GEN_STAT_MSG) LB.

ORPGVST_read() is used to read the Volume Status message from the General
Status Message LB.  Normally, this function is invoked internally when a call
is made to one of the get functions and an internal flag is set indicating the
message needs to be initialized or updated.

If this function is called externally, the argument passed can either be NULL
or a valid address to a Vol_stat_gsm_t data structure.  If the argument is NULL, 
this function allocates storage for a Vol_stat_gsm_t structure to hold the Volume
Status.  On success, the address of the Vol_stat_gsm_t data structure holding
the Volume Status data is returned.  On failure, NULL is returned.

If subsequent calls to this function are made, the user should provide the 
returned address from the first invocation.

ORPGVST_io_status() returns the status of the last Volume Status message read
operation.

ORPGVST_status_update_flag() returns 0 if the Volume Status message has been
updated since the last read, otherwise it returns 1.  One would normally use
this in an application to determine if any local data need to be updated.

ORPGVST_status_update_time() returns the time (in julian seconds) of the last
Volume Status message update.

ORPGVST_get_volume_number() returns the current volume sequence number.  This
number is initially set to 0 and increases monotonically with each new volume.
On failure, ORPGVST_DATA_NOT_FOUND is returned.

ORPGVST_get_volume_scan() returns the current volume scan number.  This
number is initially set to 0 but is otherwise a number in [1, 80].  The
volume scan number is the modulo( volume sequence number, 80) + 1. 
On failure, ORPGVST_DATA_NOT_FOUND is returned.

ORPGVST_get_volume_time() returns the current volume scan time (in milliseconds
past midnight).  On failure, ORPGVST_DATA_NOT_FOUND is returned.

ORPGVST_get_volume_flag() returns value indicating whether the current volume
is the first volume.  If the value is set, then the current volume is the
initial volume.  If the value is not set (0), then it is not the initial
volume.  On failure,  ORPGVST_DATA_NOT_FOUND is returned.

ORPGVST_get_volume_duration() returns the expected volume duration (in minutes)
from the time it started.  On failure, ORPGVST_DATA_NOT_FOUND is returned.

ORPGVST_get_volume_date() returns the current volume scan julian date.  On
failure, ORPGCST_DATA_NOT_FOUND is returned.

ORPGVST_get_previous_status() returns a value indicating the status of the
previous volume scan.  If the value is 0 (ORPGVST_ABORTED), the previous volume
scan was aborted.  If the value is 1 (ORPGVST_SUCCESS), the previous volume
scan was successful.  On failure, ORPGVST_DATA_NOT_FOUND is returned.

ORPGVST_get_mode() returns the current operational mode.  The valid modes are: 

		ORPGVST_PRECIPITATION_MODE,
		ORPGVST_CLEAR_AIR_MODE

On failure, ORPGVST_DATA_NOT_FOUND not found.

ORPGVST_get_vcp() returns the VCP number associated with the current volume
scan.  On failure, ORPGVST_DATA_NOT_FOUND is returned.

ORPGVST_get_vcp_id() returns the index of the VCP in the table of VCP data for
the VCP associated with the current volume scan.  On failure, ORPGVST_DATA_NOT_FOUND 
is returned.

ORPGVST_get_number_elevations() returns the number of unique elevation cuts in
the current VCP.  On failure, ORPGVST_DATA_NOT_FOUND is returned.

ORPGVST_get_elevation() returns the scaled (deg*10) elevation angle for the
specified elevation index (indx) in the current VCP.  On failure,
ORPGVST_DATA_NOT_FOUND is returned.

ORPGVST_get_index() returns the RPG index number associated with the specified
elevation index (indx) in the current VCP.  On failure, ORPGVST_DATA_NOT_FOUND
is returned.

ORPGVST_get_rda_index() returns the RDA index number associated with the specified
RPG elevation index (indx) in the current VCP.  On failure, ORPGVST_DATA_NOT_FOUND
is returned.

ORPGVST_get_scan_count() returns the number of start of elevation events
detected since the start of the current volume scan.  It is set to 1 at the
start of the volume.  It is initially 0.  On failure, ORPGVST_DATA_NOT_FOUND
is returned.

ORPGVST_get_current_vcp() returns the current VCP data in the caller supplied 
buffer.  On failure, ORPGVST_DATA_NOT_FOUND is returned.

ORPGVST_get_superres_bitmap() returns the Super Resolution bitmap stored in 
the volume status.  Each bit corresponds to an RPG elevation cut with the 
least significant bit being the first cut.  If the bit is set, the cut is 
Super Resolution.  On failure, ORPGVST_DATA_NOT_FOUND is returned.

ORPGVST_get_dual_pol_expected() returns 1 if Dual Pol data is expected with
this VCP, or 0 if Dual Pol data is not expected.   On failure, 
ORPGVST_DATA_NO_FOUND is returned.

ORPGVST_is_rpg_elev_superres() returns 1 if the RPG elevation index (indx) 
is a Super Resolution cut or 0 if the RPG elevation is not a Super Resolution
cut.  On failure, ORPGVST_DTAT_NOT_FOUND is returned.

ORPGVST_get_vcp_supplenmental_data() returns the VCP supplemental data word 
from the volume status.  Bit definitions can be found in vcp.h.  On failure,
ORPGVST_DATA_NOT_FOUND is returned.

ORPGVST_get_n_sails_cuts() returns the number of SAILS cuts in the current
VCP.  On failure, ORPGVST_DATA_NOT_FOUND is returned.  
 
ORPGVST_get_sails_cut_seq() returns the SAILS cut sequence number for the
specified elevation index (indx) in the current VCP.  On failure,
ORPGVST_DATA_NOT_FOUND is returned.  Sequence numbers start with 1.  The
lowest cut of the VCP is not considered a SAILS cut.  If the indx corresponds
to a non-SAILS cut, 0 is returned.
 

EXAMPLES

The following code segment is used to retrieve the current ORPG operational
mode.

	#include <orpg.h>
	#include <orpgvst.h>
	.
	.
	int	mode;
	.
	.
	mode = ORPGVST_get_mode ();
	.
	.

SEE ALSO

orpg(3)


				$Revision: 1.10 $


files_orpg_sw/src/cpc101/lib003/orpgvst.c
/*
 * RCS info
 * $Author: steves $
 * $Locker:  $
 * $Date: 2014/08/19 21:05:52 $
 * $Id: orpgvst.c,v 1.20 2014/08/19 21:05:52 steves Exp $
 * $Revision: 1.20 $
 * $State: Exp $
 */

/************************************************************************
 *									*
 *	Module:  orpgvst.c						*
 *									*
 *	Description:  This module contains a collection of functions	*
 *		      to manipulate Volume Status info message in 	*
 *		      the gen_stat_msg linear buffer.			*
 *									*
 ************************************************************************/

#include <orpg.h>
#include <orpgvst.h>
#include <orpgevt.h>

/*	Static variables						*/

static	Vol_stat_gsm_t	*Scan_info                   = NULL;
static	int		Init_vol_status_flag         = 0;
static	int		Vol_io_status                = 0;
static	int		ORPGVST_status_registered    = 0;
static	time_t		VST_status_update_time       = 0;

/************************************************************************
 *									*
 *	Description: This function returns the status of the last	*
 *		     Volume Status msg I/O operation.			*
 *									*
 ************************************************************************/
int ORPGVST_io_status(){

	return (Vol_io_status);

/* End of ORPGVST_io_status() */
}

/************************************************************************
 *									*
 *	Description: This function returns 0 if the Volume Status msg	*
 *		     needs to be updated, otherwise 1.			*
 *									*
 ************************************************************************/
int ORPGVST_status_update_flag(){

	return (Init_vol_status_flag);

/* End of ORPGVST_status_update_flag() */
}

/************************************************************************
 *									*
 *	Description: This function returns the time (in julian seconds)	*
 *	when the Volume Status message was last updated.		*
 *									*
 ************************************************************************/
time_t ORPGVST_status_update_time(){

	return (VST_status_update_time);

/* End of ORPGVST_status_update_time() */
}

/************************************************************************
 *                                                                      *
 *      Description: The following routine sets the Volume Status       *
 *                   msg init flag whenever Volume Status message       *
 *                   LB notification is received.                       *
 *                                                                      *
 *      Return:      NONE                                               *
 *                                                                      *
 ************************************************************************/
void ORPGVST_lb_notify_callback( int fd, LB_id_t msgid, int msginfo, 
                                 void *arg ){
 
        Init_vol_status_flag = 0;

/* End of ORPGVST_lb_notify_callback() */
}

/************************************************************************
 *									*
 *	Description: The following routine reads the Volume Status msg	*
 *		     (VOL_STAT_GSM_ID) from the general status message	*
 *		     lb (ORPGDAT_GEN_STAT_MSG).				*
 *									*
 *	Return:      A pointer to the Volume Status Data or NULL on	*
 *		     error.						*
 *									*
 ************************************************************************/
char*  ORPGVST_read ( char* scan_info ){

	int	status;

/*      Check to see if Volume Status message LB notification has been  *
 *      registered.  If not, register it.                               */

        if (!ORPGVST_status_registered) {

            if( ((status = ORPGDA_write_permission( ORPGDAT_GSM_DATA )) < 0)
                                    ||
                ((status = ORPGDA_UN_register( ORPGDAT_GSM_DATA,
                                               VOL_STAT_GSM_ID, 
                                               ORPGVST_lb_notify_callback)) < 0) ){

                LE_send_msg (GL_CONFIG,
                        "ORPGVST: ORPGDA_UN_register failed (ret %d)\n", status);
                return (NULL);

            }

            ORPGVST_status_registered = 1;

        }

	Init_vol_status_flag = 1;

/*	If the passed argument is not NULL, set Scan_info.  Otherwise	*
        allocate Scan_info. 						*/
        if( scan_info != NULL ){

           Scan_info = (Vol_stat_gsm_t *) scan_info;
	   status = ORPGDA_read( ORPGDAT_GSM_DATA, Scan_info,
			         sizeof (Vol_stat_gsm_t), VOL_STAT_GSM_ID );

        }
        else{

           if( Scan_info == NULL )
	      status = ORPGDA_read( ORPGDAT_GSM_DATA, (char *) &Scan_info,
	   		            LB_ALLOC_BUF, VOL_STAT_GSM_ID );

           else{

              LE_send_msg( GL_ERROR, "Potential Memory Leak in ORPGVST_read()\n" );
	      status = ORPGDA_read( ORPGDAT_GSM_DATA, (char *) Scan_info,
	   		            sizeof(Vol_stat_gsm_t), VOL_STAT_GSM_ID );

           }

        }

	Vol_io_status = status;

	if (status < 0) {

	    Init_vol_status_flag = 0;
	    LE_send_msg (GL_INFO, "ORPGDA_read (VOL_STAT_GSM_ID): %d\n", status);
            Scan_info = NULL;

	}
        else 
	    VST_status_update_time = time (NULL);

	return( (char *) Scan_info );

/* End of ORPGVST_read () */
}

/************************************************************************
 *									*
 *	Description: The following function returns the volume sequence	*
 *		     number.						*
 *									*
 *	Return:      On failure, ORPGVST_DATA_NOT_FOUND is returned.	*
 *									*
 ************************************************************************/
unsigned long ORPGVST_get_volume_number (){

	if (!Init_vol_status_flag) {

	    if( ORPGVST_read( (char *) Scan_info ) == NULL )
		return( (unsigned long) ORPGVST_DATA_NOT_FOUND );

	}

	return Scan_info->volume_number;

/* End of ORPGVST_get_volume_number() */
}

/************************************************************************
 *                                                                      *
 *      Description: The following function returns the volume scan     *
 *                   number.                                            *
 *                                                                      *
 *      Return:      On failure, ORPGVST_DATA_NOT_FOUND is returned.    *
 *                                                                      *
 ************************************************************************/
int ORPGVST_get_volume_scan(){

        if (!Init_vol_status_flag) {

	    if( ORPGVST_read( (char *) Scan_info ) == NULL )
		return( (int) ORPGVST_DATA_NOT_FOUND );

        }

        return Scan_info->volume_scan;

/* End of ORPGVST_get_volume_scan() */
}

/************************************************************************
 *									*
 *	Description: The following function returns the current volume	*
 *		     scan time (milliseconds past midnight).		*
 *									*
 *	Return:      On failure, ORPGVST_DATA_NOT_FOUND is returned.	*
 *									*
 ************************************************************************/
unsigned long ORPGVST_get_volume_time(){

	if (!Init_vol_status_flag) {

	    if( ORPGVST_read( (char *) Scan_info ) == NULL )
		return( (unsigned long) ORPGVST_DATA_NOT_FOUND );

	}

	return Scan_info->cv_time;

/* End of ORPGVST_get_volume_time() */
}

/************************************************************************
 *									*
 *	Description: The following function returns the value of the	*
 *		     initial volume flag.				*
 *									*
 *	Return:      On failure, ORPGVST_DATA_NOT_FOUND is returned.	*
 *									*
 ************************************************************************/
int ORPGVST_get_volume_flag(){

	if (!Init_vol_status_flag) {

	    if( ORPGVST_read( (char *) Scan_info ) == NULL )
		return( (int) ORPGVST_DATA_NOT_FOUND );
	     
	}

	return (int) Scan_info->initial_vol;

/* End of ORPGVST_get_volume_flag() */
}

/************************************************************************
 *									*
 *	Description: The following function returns the expected	*
 *		     volume duration (seconds).				*
 *									*
 *	Return:      On failure, ORPGVST_DATA_NOT_FOUND is returned.	*
 *									*
 ************************************************************************/
int ORPGVST_get_volume_duration(){

	if (!Init_vol_status_flag) {

	    if( ORPGVST_read( (char *) Scan_info ) == NULL )
		return( (int) ORPGVST_DATA_NOT_FOUND );
	     
	}

	return (int) Scan_info->expected_vol_dur;

/* End of ORPGVST_get_volume_duration() */
}

/************************************************************************
 *									*
 *	Description: The following function returns the current volume	*
 *		     scan julian date.					*
 *									*
 *	Return:      On failure, ORPGVST_DATA_NOT_FOUND is returned.	*
 *									*
 ************************************************************************/
int ORPGVST_get_volume_date (){

	if (!Init_vol_status_flag) {

	    if( ORPGVST_read( (char *) Scan_info ) == NULL )
		return( (int) ORPGVST_DATA_NOT_FOUND );
	     
	}

	return (int) Scan_info->cv_julian_date;

/* End of ORPGVST_get_volume_date() */
}

/************************************************************************
 *									*
 *	Description: The following function returns the previous	*
 *		     volume status.					*
 *									*
 *	Return:      On failure, ORPGVST_DATA_NOT_FOUND is returned.	*
 *		     On success, ORPGVST_ABORTED or ORPGVST_SUCCESS	*
 *		     are returned.					*
 *									*
 ************************************************************************/
int ORPGVST_get_previous_status (){

	if (!Init_vol_status_flag) {

	    if( ORPGVST_read( (char *) Scan_info ) == NULL )
		return( (int) ORPGVST_DATA_NOT_FOUND );
	     
	}

	return (int) Scan_info->pv_status;

/* End of ORPGVST_get_previous_status() */
}

/************************************************************************
 *									*
 *	Description: The following function returns the current mode	*
 *		     of operation.					*
 *									*
 *	Return:      On failure, ORPGVST_DATA_NOT_FOUND is returned.	*
 *		     On success, ORPGVST_PRECIPITATION_MODE, or		*
 *		     ORPGVST_CLEAR_AIR_MODE is returned.		*
 *									*
 ************************************************************************/
int ORPGVST_get_mode(){

	if (!Init_vol_status_flag) {

	    if( ORPGVST_read( (char *) Scan_info ) == NULL )
		return( (int) ORPGVST_DATA_NOT_FOUND );

	}

	return (int) Scan_info->mode_operation;

/* End of ORPGVST_get_mode() */
}

/************************************************************************
 *									*
 *	Description: The following function returns the volume coverage	*
 *		     pattern (VCP) in use.				*
 *									*
 *	Return:      On failure, ORPGVST_DATA_NOT_FOUND is returned.	*
 *									*
 ************************************************************************/
int ORPGVST_get_vcp(){

	if (!Init_vol_status_flag) {

	    if( ORPGVST_read( (char *) Scan_info ) == NULL )
		return( (int) ORPGVST_DATA_NOT_FOUND );
	     
	}

	return (int) Scan_info->vol_cov_patt;

/* End of ORPGVST_get_vcp() */
}

/************************************************************************
 *                                                                      *
 *      Description: The following function returns the volume coverage *
 *                   pattern (VCP) index into the VCP definition table  *
 *                   of the VCP currently in use.                       *
 *                                                                      *
 *      Return:      On failure, ORPGVST_DATA_NOT_FOUND is returned.    *
 *                                                                      *
 ************************************************************************/
int ORPGVST_get_vcp_id(){

        if (!Init_vol_status_flag) {

	    if( ORPGVST_read( (char *) Scan_info ) == NULL )
		return( (int) ORPGVST_DATA_NOT_FOUND );
             
        }

        return (int) Scan_info->rpgvcpid;

/* End of ORPGVST_get_vcp_id() */
}

/************************************************************************
 *									*
 *	Description: The following function returns the number of	*
 *		     elevation cuts in the current VCP.			*
 *									*
 *	Return:      On failure, ORPGVST_DATA_NOT_FOUND is returned.	*
 *									*
 ************************************************************************/
int ORPGVST_get_number_elevations(){

	if (!Init_vol_status_flag) {

	    if( ORPGVST_read( (char *) Scan_info ) == NULL )
		return( (int) ORPGVST_DATA_NOT_FOUND );
	     
	}

	return (int) Scan_info->num_elev_cuts;

/* End of ORPGVST_get_number_elevations() */
}

/************************************************************************
 *									*
 *	Description: The following function returns the	elevation	*
 *		     angle (deg*10) for the specified elev index. 	*
 *									*
 *	Return:      On failure, ORPGVST_DATA_NOT_FOUND is returned.	*
 *									*
 *	Note:	     The "indx" is the RDA elevation index - 1.		*
 *									*
 ************************************************************************/
int ORPGVST_get_elevation( int indx ){
 
	if (!Init_vol_status_flag) {

	    if( ORPGVST_read( (char *) Scan_info ) == NULL )
		return( (int) ORPGVST_DATA_NOT_FOUND );

	}

	if ((indx < 0) ||
	    (indx >= Scan_info->num_elev_cuts)) {

	    return (int) ORPGVST_DATA_NOT_FOUND;

	} else {

	    return (int) Scan_info->elevations [indx];

	}

/* End of ORPGVST_get_elevation() */
}

/************************************************************************
 *									*
 *	Description: The following function returns the	RPG elevation	*
 *		     index associated with the corresponding elev	*
 *		     index.						*
 *									*
 *	Return:      On failure, ORPGVST_DATA_NOT_FOUND is returned.	*
 *									*
 *	Note:        The "indx" is the RDA elevation index - 1.		*
 *									*
 ************************************************************************/
int ORPGVST_get_index( int indx ){ 

	if (!Init_vol_status_flag) {

	    if( ORPGVST_read( (char *) Scan_info ) == NULL )
		return( (int) ORPGVST_DATA_NOT_FOUND );

	}

	if ((indx < 0) ||
	    (indx >= Scan_info->num_elev_cuts)) {

	    return (int) ORPGVST_DATA_NOT_FOUND;

	} else {

	    return (int) Scan_info->elev_index [indx];

	}

/* End of ORPGVST_get_index() */
}

/************************************************************************
 *									*
 *	Description: The following function returns the	RDA elevation	*
 *		     index associated with the corresponding indx.      *
 *									*
 *	Return:      On failure, ORPGVST_DATA_NOT_FOUND is returned.	*
 *									*
 *	Note:        This is a worthless function and should be 	*
 *		     deprecated.					*
 *									*
 ************************************************************************/
int ORPGVST_get_rda_index( int indx ){

	int status, i;

	if (!Init_vol_status_flag) {

	    if( ORPGVST_read( (char *) Scan_info ) == NULL )
		return( (int) ORPGVST_DATA_NOT_FOUND );
	     
	}

        status = -1;
        for (i = 0; (i < Scan_info->num_elev_cuts); i++)
        {
            if (Scan_info->elev_index[i] == indx)
            {
                status = i;
                break;
            }
        }
	if (status == -1)
           status = ORPGVST_DATA_NOT_FOUND;
	return (status);

/* End of  ORPGVST_get_rda_index() */
}

/************************************************************************
 *									*
 *	Description: The following function returns the	current VCP  	*
 *		     table in the user supplied buffer.                 *
 *									*
 *	Return:      On failure, ORPGVST_DATA_NOT_FOUND is returned.	*
 *									*
 ************************************************************************/
int ORPGVST_get_current_vcp( Vcp_struct *vcp ){

        if (!Init_vol_status_flag) {

	    if( ORPGVST_read( (char *) Scan_info ) == NULL )
		return( (int) ORPGVST_DATA_NOT_FOUND );
             
        }

        if( vcp == NULL ) 
            return (int) ORPGVST_DATA_NOT_FOUND;

        memcpy( vcp, &Scan_info->current_vcp_table, sizeof(Vcp_struct) );

        return(0);

/* End of ORPGVST_get_current_vcp() */
}

/************************************************************************
 *                                                                      *
 *      Description: The following function returns the expected super  *
 *                   res bit map.					*
 *                                                                      *
 *      Return:      On failure, ORPGVST_DATA_NOT_FOUND is returned.    *
 *                                                                      *
 ************************************************************************/
int ORPGVST_get_superres_bitmap( ){

        if (!Init_vol_status_flag) {

            if( ORPGVST_read( (char *) Scan_info ) == NULL )
                return( (int) ORPGVST_DATA_NOT_FOUND );

        }

        return( Scan_info->super_res_cuts );

/* End of ORPGVST_get_superres_bitmap() */
}

/************************************************************************
 *                                                                      *
 *      Description: The following function returns the Dual Pol        *
 *                   expected value.		*
 *                                                                      *
 *      Return:      On failure, ORPGVST_DATA_NOT_FOUND is returned.    *
 *                                                                      *
 ************************************************************************/
int ORPGVST_get_dual_pol_expected(){

        if (!Init_vol_status_flag) {

            if( ORPGVST_read( (char *) Scan_info ) == NULL )
                return( (int) ORPGVST_DATA_NOT_FOUND );

        }

        return( Scan_info->dual_pol_expected );

/* End of ORPGVST_get_dual_pol_expected() */
}

/************************************************************************
 *                                                                      *
 *      Description: The following function returns 1 (super res) or    *
 *                   0 (not super res) based on rpg elevation index.	*
 *                                                                      *
 *      Return:      On failure, ORPGVST_DATA_NOT_FOUND is returned.    *
 *                                                                      *
 *	Notes:	     The rpg elevation number begins with 1. The bit	*
 *		     map uses bit 0 to denote elevation 1.		*
 *									*
 ************************************************************************/
int ORPGVST_is_rpg_elev_superres( int rpg_elev_num ){

	int bit;

        if (!Init_vol_status_flag) {

            if( ORPGVST_read( (char *) Scan_info ) == NULL )
                return( (int) ORPGVST_DATA_NOT_FOUND );

        }

        bit = 1 << (rpg_elev_num - 1);
        if( Scan_info->super_res_cuts & bit )
           return 1;

        return( 0 );

/* End of ORPGVST_is_rpg_elev_superres() */
}

/************************************************************************
 *                                                                      *
 *      Description: The following function returns the VCP Supplemental*
 *                   bitmap.            			 	*
 *                                                                      *
 *      Return:      On failure, ORPGVST_DATA_NOT_FOUND is returned.    *
 *                                                                      *
 ************************************************************************/
int ORPGVST_get_vcp_supplemental_data(){

        if (!Init_vol_status_flag) {

            if( ORPGVST_read( (char *) Scan_info ) == NULL )
                return( (int) ORPGVST_DATA_NOT_FOUND );

        }

        return( Scan_info->vcp_supp_data );

/* End of ORPGVST_get_vcp_supplemental_data() */
}

/************************************************************************
 *									*
 *	Description: The following function returns the	number of SAILS *
 *		     cuts in the current vcp.                    	*
 *									*
 *	Return:      On failure, ORPGVST_DATA_NOT_FOUND is returned.	*
 *									*
 ************************************************************************/
int ORPGVST_get_n_sails_cuts(){ 

	if (!Init_vol_status_flag) {

	    if( ORPGVST_read( (char *) Scan_info ) == NULL )
		return( (int) ORPGVST_DATA_NOT_FOUND );

	}

	return (int) Scan_info->n_sails_cuts;

/* End of ORPGVST_get_n_sails_cuts() */
}

/************************************************************************
 *                                                                      *
 *      Description: The following function returns the SAILS cut       *
 *                   sequence number for the cut, or 0 if not a SAILS	*
 *                   cut. 						*
 *                                                                      *
 *      Return:      On failure, ORPGVST_DATA_NOT_FOUND is returned.    *
 *                                                                      *
 *      Note:        The "indx" is the RDA elevation index - 1.         *
 *                                                                      *
 ************************************************************************/
int ORPGVST_get_sails_cut_seq( int indx ){
 
        if (!Init_vol_status_flag) {

            if( ORPGVST_read( (char *) Scan_info ) == NULL )
                return( (int) ORPGVST_DATA_NOT_FOUND );

        }

        if ((indx < 0) ||
            (indx >= Scan_info->num_elev_cuts)) {

            return (int) ORPGVST_DATA_NOT_FOUND;

        } else {

            return (int) Scan_info->sails_cut_seq [indx];

        }

/* End of ORPGVST_get_sails_cut_seq() */
}

